dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/DefaultTrackerObjectsDeletionService.java

```diff
@@ -195,17 +199,19 @@ public TrackerTypeReport deleteTrackedEntities(List<String> trackedEntities)
 
       Set<Enrollment> daoEnrollments = entity.getEnrollments();
 
-      List<String> enrollments =
+      List<UID> enrollments =
           daoEnrollments.stream()
               .filter(enrollment -> !enrollment.isDeleted())
               .map(BaseIdentifiableObject::getUid)
+              .map(UID::of)
               .toList();
       deleteEnrollments(enrollments);
 
       RelationshipQueryParams params = RelationshipQueryParams.builder().entity(entity).build();
-      List<String> relationships =
+      List<UID> relationships =
           relationshipStore.getByTrackedEntity(entity, params).stream()
               .map(BaseIdentifiableObject::getUid)
```

Comment:
UID::of can take an IdentifiableObject right? if so we don't need this mapping. There are more such cases in this PR.